### PR TITLE
Keras 3 persistence: native .keras save/load, round‑trip tests, custom layer serialization, CI

### PR DESCRIPTION
**Pull request by Thunderblok**
Summary
Introduces robust Keras 3 persistence: native .keras save/load with safe_mode and sibling .meta.json.
Adds deterministic seeding utility for reproducible runs.
Implements round‑trip tests for base and registered custom layers; CPU‑only for stability.
What’s included
Helpers: save_model_and_meta, load_model_safe, load_meta (cerebros/persistence/keras_io.py).
Seeding: set_global_seed (cerebros/utils/random.py).
Tests:
test_keras_persistence_roundtrip.py (pred parity + resume training equality + meta file).
test_keras_custom_layer_serialization.py (registered custom layer; no custom_objects required).
CI: persist.yml (Py 3.10; numpy==1.26.4; tensorflow==2.20.0; pytest; runs both tests).
Audit: audit_serialization.py (flags unregistered Layer subclasses).
Why it matters
Save → load → validate is now turnkey and safe for custom layers.
Resume training picks up optimizer state consistently.
Validation
Local: both tests pass.
CI: green on fork.
Notes
NumPy 2.x may emit a harmless array(copy=...) deprecation; CI pins numpy==1.26.4 to avoid it.
Tests force CPU to avoid CUDA driver/arch variance.
Next (non‑blocking)
Negative test for unregistered layers (expect safe_mode failure with clear error).
Multi‑IO + BatchNorm round‑trip.
Explicit optimizer.iterations and optimizer weights assertions.
Idempotency: save→load→save→load parity.
Run the audit script in CI; optional TF 2.19/2.20 matrix; SavedModel parity.

### CODE REVIEW

**Comment by david-thrower** [comment]
@Thunderblok  I have an important question:

cerebros/keras_export.py lines 74 through 76


```
        # Deduplicate while preserving order
        seen = set()
        inputs = [x for x in inputs if not (x in seen or seen.add(x))]
```

If I am following this right, if I had the graph:

```
inp = Input()
dense_level1_unit1 = Dense(2, 'gelu')(inp)
dense_level1_unit2 = Dense(5, 'gelu' )(inp,inp,inp)(inp)

### Pay attention to this node <<<<-------------------------------------<<<<<<<<<<<<
cat_level_2_unit_1 = Concatenate(axis=1)(
    [ inp,
      inp, 
      dense_level1_unit1,
      dense_level1_unit1, 
      dense_level1_unit2, 
      dense_level1_unit2, 
])
dense_level_2_unit_1 =  Dense(5, 'gelu' )(cat_level_2_unit_1)
```

### And this were capturing the upstream connectivity of `cat_level_2_unit_1`
... 

The node `cat_level_2_unit_1` has intentional duplication, the concatenation of `[inp, inp, dense_level1_unit1, dense_level1_unit1, dense_level1_unit2,  dense_level1_unit2  ]`
...

Would this be searilizing this node as  `[inp, dense_level1_unit1, dense_level1_unit2  ]`? If so, this is not correct, as the Cerebros graph has a hyperdesnse-sparse lattice of connectivity in all layers and should not be de-duplicated. If I am misreding this, hopefully you or the AI can elaborate where I am missing this.
